server/src/test/java/org/elasticsearch/cluster/metadata/IndexMetadataTests.java

```diff
@@ -309,36 +309,36 @@ public void testGetRoutingFactor() {
     public void testSelectShrinkShards() {
         int numberOfReplicas = randomIntBetween(0, 10);
         IndexMetadata metadata = IndexMetadata.builder("foo")
-            .settings(indexSettings(32, numberOfReplicas).put("index.version.created", 1))
```

Comment:
I think we need to retain at least a minimal version of the test for the old index version.